### PR TITLE
Fix a bug in TranslateWin32Expression

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Add test coverage.

cc: @danmosemsft, @Anipik, @pjanotti

### CODE REVIEW

File: src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
+            FileInfo fileOne = new FileInfo(Path.Combine(testDirectory.FullName, "a.one"));
+            FileInfo fileTwo = new FileInfo(Path.Combine(testDirectory.FullName, "ab.two"));
+            FileInfo fileThree = new FileInfo(Path.Combine(testDirectory.FullName, "abc.three"));
+
+            fileOne.Create().Dispose();
+            fileTwo.Create().Dispose();
+            fileThree.Create().Dispose();
+
+            // Question marks collapse to periods in Win32
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs]: Period in what sense? Not in the regex sense of if can match nothing

File: src/System.IO.FileSystem/tests/System.IO.FileSystem.Tests.csproj [hunk 2]
```diff
@@ -195,8 +196,6 @@
   <ItemGroup>
     <EmbeddedResource Include="Resources\$(AssemblyName).rd.xml" />
   </ItemGroup>
-  <ItemGroup>
-    <Folder Include="Matchers\" />
-  </ItemGroup>
+  <ItemGroup />
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/System.IO.FileSystem.Tests.csproj]: Nit, Redundant

File: src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs]: Literal period

File: src/System.IO.FileSystem/tests/System.IO.FileSystem.Tests.csproj (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/System.IO.FileSystem.Tests.csproj]: Egh- clearly I didn't add that explicitly. :)

File: src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs (hunk 1 above)
Comment (danmosemsft) [reply src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs]: I guess I don't know what collapse to period means.

Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs]: It means that question marks are optional if they but up contiguously against a period.  So a???.foo is actually a.foo | a?.foo | a??.foo | a???.foo. Evil that is attempted 8.3 matching matching. :P

File: src/System.IO.FileSystem/tests/Enumeration/Win32MatcherTests.netcoreapp.cs [hunk 3]
```diff
@@ -112,5 +129,19 @@ public static void Win32Match(string expression, string name, bool ignoreCase, b
             { @"<.", @"a.", true, true },
             { @"<.", @"a.b", true, false },
         };
+
+        [Theory,
+            InlineData("", "*"),
```
Comment (ahsonkhan) [inline src/System.IO.FileSystem/tests/Enumeration/Win32MatcherTests.netcoreapp.cs]: That's an interesting way of specifying the test data. I didn't know you could do that.

I have almost always seen it as:
```C#
[Theory]
[InlineData("", "*")]
[InlineData("*.*", "*")]
...
public void TranslateExrpression(string expression, string expected)
```

**Pull request merged**